Analysis/src/BaseEventAnalyzer.cpp

```diff
@@ -271,8 +272,12 @@ void BaseEventAnalyzer::ProcessDataSource(const SampleDescriptor& sample, const
 
                 if(ana_setup.period == Period::Run2016 || ana_setup.period == Period::Run2017)
                     l1_prefiring_weight = (*event)->l1_prefiring_weight;
-            }
 
+                auto jet_pu_id_weight_provided = eventWeights_HH->GetProviderT<mc_corrections::JetPuIdWeights>(
+                        mc_corrections::WeightType::JetPuIdWeights);
+                jet_pu_id_weight = jet_pu_id_weight_provided->Get(*event);
```

Comment:
total weight should include `jet_pu_id_weight`. (if needed, set it to 1 in your local copy for the sync)